### PR TITLE
Difficulty

### PR DESCRIPTION
**Pull request by maksymivv**

### CODE REVIEW

File: src/CryptoNoteConfig.h [hunk 1]
```diff
@@ -69,7 +71,7 @@ const size_t   FUSION_TX_MAX_SIZE                            = CRYPTONOTE_BLOCK_
 const size_t   FUSION_TX_MIN_INPUT_COUNT                     = 12;
 const size_t   FUSION_TX_MIN_IN_OUT_COUNT_RATIO              = 4;
 
-const uint32_t UPGRADE_HEIGHT_V2                                = 4294967294;
+const uint32_t UPGRADE_HEIGHT_V2                                = 60000;
 const uint32_t UPGRADE_HEIGHT_V3                                = 4294967295;
```
Comment (pmitchev) [inline src/CryptoNoteConfig.h]: change upgrade_height_v3 to 4294967294 or you will go to major_version = 3 when 90% of the mined blocks use this version

File: src/CryptoNoteConfig.h [hunk 2]
```diff
@@ -48,8 +49,9 @@ const uint64_t MINIMUM_FEE                                   = UINT64_C(10000000
 const uint64_t DEFAULT_DUST_THRESHOLD                        = UINT64_C(100000000);
 
 const uint64_t DIFFICULTY_TARGET                             = 240; // seconds
-const uint64_t EXPECTED_NUMBER_OF_BLOCKS_PER_DAY             = 360;
+const uint64_t EXPECTED_NUMBER_OF_BLOCKS_PER_DAY             = 24 * 60 * 60 / DIFFICULTY_TARGET;
 const size_t   DIFFICULTY_WINDOW                             = EXPECTED_NUMBER_OF_BLOCKS_PER_DAY; // blocks
+const size_t   DIFFICULTY_WINDOW_V2                          = 17; // blocks
```
Comment (pmitchev) [inline src/CryptoNoteConfig.h]: is the DIFFICULTY_WINDOW_V2 the only change in the difficulty calculation? (just to know should I read all the code about the diff calculation)

Comment (maksymivv) [inline src/CryptoNoteConfig.h]: There are changes in Difficulty.cpp, Difficulty.h, Currency.cpp, Currency.h to make another diff algo for blocks >= V2

Comment (maksymivv) [inline src/CryptoNoteConfig.h]: This request is by mistake :) i will make another one

**Pull request closed without merging**